### PR TITLE
Add VK_QCOM_image_processing

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Rebase of https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4611

Adds `VK_QCOM_image_processing`

### CODE REVIEW

**Review by juan-lunarg — APPROVED** [review]

File: tests/negative/image.cpp [hunk 1]
```diff
@@ -6341,3 +6341,207 @@ TEST_F(VkLayerTest, SlicedCreateInfoInvalidUsage) {
...
+                      {3, VK_DESCRIPTOR_TYPE_BLOCK_MATCH_IMAGE_QCOM, 1, VK_SHADER_STAGE_FRAGMENT_BIT, nullptr},
+                  });
+    descriptor_set.WriteDescriptorImageInfo(0, sampled_image_view, VK_NULL_HANDLE, VK_DESCRIPTOR_TYPE_SAMPLE_WEIGHT_IMAGE_QCOM);
+    descriptor_set.WriteDescriptorImageInfo(3, weight_image_view, VK_NULL_HANDLE, VK_DESCRIPTOR_TYPE_BLOCK_MATCH_IMAGE_QCOM);
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkWriteDescriptorSet-descriptorType-06942");
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkWriteDescriptorSet-descriptorType-06943");
+    descriptor_set.UpdateDescriptorSets();
+    m_errorMonitor->VerifyFound();
+
+    vk::DestroyImageView(m_device->device(), weight_image_view, NULL);
```
Comment (arno-lunarg) [inline tests/negative/image.cpp:6544]: Nit: Use `vk_testing::ImageView` and `vk_testing::Sampler` to enjoy RAII?

**Review by arno-lunarg — APPROVED** [review]
LGTM! One typo and one suggestion

File: tests/negative/image.cpp [hunk 2]
```diff
@@ -6341,3 +6341,207 @@ TEST_F(VkLayerTest, SlicedCreateInfoInvalidUsage) {
         m_errorMonitor->VerifyFound();
     }
 }
+
+TEST_F(VkLayerTest, ImageViewTextureSampleWeighted) {
```
Comment (juan-lunarg) [inline tests/negative/image.cpp:6345]: Nitpick FFTI

Add a positive for VK_QCOM_image_processing

**Review by ncesario-lunarg — COMMENTED** [review]
LGTM once the comments are resolved!

File: tests/negative/image.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/negative/image.cpp:6544]: agree, I just copied the test over, happy to improve it a bit now

File: tests/negative/image.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/negative/image.cpp:6345]: I would, but not what a "good positive" would be (know nothing about the extension use case), I could write a bunch of things that are positive, that wouldn't test it well

**Pull request merged**